src/signaler.cpp

```diff
@@ -239,8 +239,17 @@ void zmq::signaler_t::send ()
 #endif
 }
 
+int zmq_poller_poll (zmq_pollitem_t *items_, int nitems_, long timeout_);
+
 int zmq::signaler_t::wait (int timeout_)
 {
+#if defined ZMQ_HAVE_POLLER
```

Comment:
I am not sure, but it zmq_poller_poll (respectively poller_t) might have not been used here intentionally. If that was the case, then probably for performance reasons. The performance impact of this should be checked.

zmq_poller_poll is an external API method, and I am also not sure if they are intended to be called from within the library.

Reply:
Actually I noticed that zmq_poller_poll is not directly an external API method,but used to implement the deprecated zmq_poll. At some point, it will be removed, so using poller_base_t sounds better anyway :)